### DISCUSSION TITLE
Unified custom route data

### DISCUSSION THREAD

**Discussion by cubxx** [hannoeru/vite-plugin-pages#589] [Ideas]
### Description

SFC custom block and JSX/TSX YAML format comments are not perfect method to custom route data, especially when I want to use imported data in route data:
```ts
import { mdiHomeOutline } from '@mdi/js';
const route = {
  meta: {
    btnProps: {
      tip: 'home',
      icon: mdiHomeOutline,
      order: 2,
      hideOn: 'mobile',
    },
  },
}
```

### Suggested solution

custom route data in runtime code is maybe better method, it has natural framework-independence advantage, and support type check:

```ts
// src/pages/index.tsx
import type { RouteMeta } from 'vue-router';
import { mdiHomeOutline } from '@mdi/js';
export const route = {
  meta: {
    btnProps: {
      tip: 'home',
      icon: mdiHomeOutline,
      order: 2,
      hideOn: 'mobile',
    } satisfies RouteMeta,
  },
}
export default ...
```

### Validations

- [X] Follow the [Code of Conduct](https://github.com/hannoeru/.github/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the docs.
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

**Comment by hannoeru** [comment]
If I add too much support to this is kind of thing which is framework specific, will cause this plugin to become hard to maintain.
I personal will recommend using some existing solution like Remix to resolve that kind of requirement.

**Comment by cubxx** [comment]
@hannoeru I don't think it is framework specific. Actually, merging custom data from page file with generated route from plugin in virtual module is feasible methods, which is what I do currently.

I used `import.meta.glob` to achieve it, just like:
```ts
// as a part of virtual module
const customs= import.meta.glob(
  ['~/pages/**/*.tsx', '!~/pages/**/components/*'],
  { import: 'route', eager: true },
);
```

of course, it will throw error when the page file don't export named `route` data, so we can add `export route = {}`  to solve it on `transform` hook.
